### ISSUE TITLE
Cannot download through mobile data

### ISSUE THREAD

**Issue by zerakori** [aniyomiorg/aniyomi#1454]
### Steps to reproduce

Open mobile data and close wifi
Disable download only on wifi option
Download episode

### Expected behavior

It should start downloading

### Actual behavior

It pauses itself and doesn't even start again no matter how many times i press resume, the button just immediately changes to show resume again

### Aniyomi version

0 15.2.4

### Android version

Android 11

### Device

Infinix note 10

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an official extension, I should be opening an issue in the [extensions repository](https://github.com/aniyomiorg/aniyomi-extensions/issues/new/choose).
- [X] If this is an issue with an official manga extension and this issue can be replicated in the Tachiyomi app, that I should be opening an issue in [Tachiyomi's extensions repository](https://github.com/tachiyomiorg/extensions/issues/new/choose).
- [X] I have gone through the [FAQ](https://aniyomi.org/docs/faq/general) and [troubleshooting guide](https://aniyomi.org/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.15.2.4](https://github.com/aniyomiorg/aniyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by fakebhai** [comment]
it's not possible to reproduce this, it's very likely specific extension problem like link getting expired. 
also if it's not starting at all (like 0%/fail) it could very well be network restrictions or cloudflare getting activated again. in any case if this happens check your webview and also try refreshing your episode list and check for any errors. and also go through [troubleshooting steps](https://aniyomi.org/help/guides/troubleshooting/#diagnosis) for more diagnosis.

And give more information on your issues otherwise we might just have to close it for lack of information.

**Comment by zerakori** [comment]
I am downloading the episodes from gogoanime which doesn't have a cloudfare, have also tried aniwatch so it isn't specific to the extension, it's not even starting to show the percentage of the episode downloaded

Uploading screen-20240221-205329.mp4…

**Comment by zerakori** [comment]
https://github.com/aniyomiorg/aniyomi/assets/85497048/4f1d41a3-7f1a-4ec5-9486-61626d7b8bd3

**Comment by fakebhai** [comment]
> 
> https://github.com/aniyomiorg/aniyomi/assets/85497048/4f1d41a3-7f1a-4ec5-9486-61626d7b8bd3
> 
> 

if you're changing it midway it's very likely not gonna work. it doesn't fetch url till it starts downloading.
when you change your network or vpn it should go through all those steps without any errors to fetch the url so whenever you change your network state make sure you do steps mentioned above:

>
>
check your webview and also try refreshing your episode list and check for any errors. and also go through [troubleshooting steps](https://aniyomi.org/help/guides/troubleshooting/#diagnosis) for more diagnosis.
>
>

to ensure it fetches everything correctly.

cancel those downloads, do mentioned steps and then start your download again.

**Comment by zerakori** [comment]
> > screen-20240221-205329.mp4
> 
> if you're changing it midway it's very likely not gonna work. it doesn't fetch url till it starts downloading. when you change your network or vpn it should go through all those steps without any errors to fetch the url so whenever you change your network state make sure you do steps mentioned above:
> 
> > 
> 
> check your webview and also try refreshing your episode list and check for any errors. and also go through [troubleshooting steps](https://aniyomi.org/help/guides/troubleshooting/#diagnosis) for more diagnosis.
> 
> > 
> 
> to ensure it fetches everything correctly.
> 
> cancel those downloads, do mentioned steps and then start your download again.

I just wanted to show it was off you can see that it was already off before starting the video

**Comment by zerakori** [comment]
Ohhhh I see where the problem occured,
I am only talking about downloading through mobile data , and not continuing the download by changing networks

**Comment by zerakori** [comment]
So can you guide what to do now

**Comment by fakebhai** [comment]
> Ohhhh I see where the problem occured,
> I am only talking about downloading through mobile data , and not continuing the download by changing networks

well it actually starts downloading for me (mostly) when I press on resume but you can try (if you want a solution/workaround) just tapping on 2 lines of extension or individual episode in `more - download queue` and it'll very likely start your download or fail it. or u can just pause and resume few more times (again possibility of download failure)

**Comment by Secozzi** [comment]
This has most likely been fixed in the [preview](https://github.com/aniyomiorg/aniyomi-preview/releases/) version with the merge from Mihon. If the issue still persists even after switching to preview or in the next stable update, please open a new issue.
